tests/nlu/classifiers/test_diet_classifier.py

```diff
@@ -609,10 +618,122 @@ async def test_train_model_checkpointing(
         - component_1_CountVectorsFeaturizer (as per the pipeline above)
         - component_2_DIETClassifier files (more than 1 file)
     """
-    all_files = list(best_model_file.rglob("*.*"))
+    all_files = list(model_dir.rglob("*.*"))
     assert len(all_files) > 4
 
 
+async def test_train_model_not_checkpointing(
+    component_builder: ComponentBuilder, tmpdir: Path, nlu_data_path: Text,
+):
+    model_name = "nlu-not-checkpointed-model"
+    checkpoint_dir = get_checkpoint_dir_path(tmpdir, Path(tmpdir, model_name))
+    assert not checkpoint_dir.is_dir()
+
+    _config = RasaNLUModelConfig(
+        {
+            "pipeline": [
+                {"name": "WhitespaceTokenizer"},
+                {"name": "CountVectorsFeaturizer"},
+                {"name": "DIETClassifier", EPOCHS: 2, CHECKPOINT_MODEL: False},
+            ],
+            "language": "en",
+        }
+    )
+
+    await rasa.nlu.train.train(
+        _config,
+        path=str(tmpdir),
+        data=nlu_data_path,
+        component_builder=component_builder,
+        fixed_model_name=model_name,
+    )
+
+    assert not checkpoint_dir.is_dir()
+
+
+async def test_train_fails_with_zero_eval_num_epochs(
+    component_builder: ComponentBuilder, tmpdir: Path, nlu_data_path: Text,
+):
+    model_name = "nlu-fail"
+    checkpoint_dir = get_checkpoint_dir_path(tmpdir, Path(tmpdir, model_name))
+    assert not checkpoint_dir.is_dir()
+
+    _config = RasaNLUModelConfig(
+        {
+            "pipeline": [
+                {"name": "WhitespaceTokenizer"},
+                {"name": "CountVectorsFeaturizer"},
+                {
+                    "name": "DIETClassifier",
+                    EPOCHS: 1,
+                    CHECKPOINT_MODEL: True,
+                    EVAL_NUM_EPOCHS: 0,
+                    EVAL_NUM_EXAMPLES: 10,
+                },
+            ],
+            "language": "en",
+        }
+    )
+    with pytest.raises(InvalidConfigException):
+        with pytest.warns(UserWarning) as warning:
+            await rasa.nlu.train.train(
+                _config,
+                path=str(tmpdir),
+                data=nlu_data_path,
+                component_builder=component_builder,
+                fixed_model_name=model_name,
+            )
+    assert not checkpoint_dir.is_dir()
+    warn_text = (
+        f"You have opted to save the best model, but the value of '{EVAL_NUM_EPOCHS}' "
+        f"is not -1 or greater than 0. Training will fail."
+    )
+    print(f"\n{warn_text}")
+    print(warning[2].message)
```

Comment:
```suggestion
```

Reply:
🤦‍♀️

Reply:
Wow, I can't believe I overlooked this 🤦